### PR TITLE
Add trento Single Sign-on integration docs

### PR DESCRIPTION
**Pull request by arbulu89**
Not to be published until version 2.4 is released.

**NOTE:**
 The xml file is created based in the migration file formatted as markdown. As `definition_lists` are not a standard component in markdown, but we already use them in the docs, i wanted to use them to list the available variables.
In order to enable the conversion of them, we need to indicate it to pandoc.
Use this command to generate the final xml:

```
pandoc --from=gfm+definition_lists+raw_attribute --to=docbook5 sso-integration.md --output ../xml/sso-integration.xml
```

### PR creator: Description

Include Trento's new single sign-on (SSO) integration usage docs.
It explains how to setup different types depending on the user needs

### PR creator: Are there any relevant issues/feature requests?

Not that I'm aware of.
@abravosuse maybe the next release jira ticket if we have one?

### CODE REVIEW

File: trento/xml/sso-integration.xml [hunk 1]
```diff
@@ -0,0 +1,762 @@
...
+  <title>Single Sign-On integration</title>
+  <para>
+    Trento can be integrated with an identity provider (IDP) that uses
+    different Single sign-on (SSO) protocols like OpenID Connect (OIDC)
+    and Open Authorization 2.0 (OAuth 2).
+  </para>
+  <blockquote>
+<literallayout>[!NOTE]
+Trento cannot start with multiple SSO options together, so only one can be chosen.</literallayout>
+  </blockquote>
```
Comment (tomschr) [inline trento/xml/sso-integration.xml:11]: This tag is wrong and we should use this:

```suggestion
<note>
  <para>Trento cannot start with multiple SSO options together, so only one can be chosen.</para>
</note>
```

File: trento/xml/sso-integration.xml [hunk 2]
```diff
@@ -0,0 +1,762 @@
...
+  <section xml:id="available-protocols">
+    <title>Available protocols</title>
+    <itemizedlist spacing="compact">
```
Comment (tomschr) [inline trento/xml/sso-integration.xml:14]: If possible, the attribute should be removed:

```suggestion
    <itemizedlist>
```

File: trento/xml/sso-integration.xml [hunk 3]
```diff
@@ -0,0 +1,762 @@
...
+      <para>
+        OIDC authentication is <emphasis role="strong">disabled by
+        default</emphasis>.
+      </para>
+      <section xml:id="enabling-oidc-when-using-rpm-packages">
+        <title>Enabling OIDC when using RPM packages</title>
+        <para>
+          Provide the following environment variables to trento-web
+          configuration, which is stored at
+          <literal>/etc/trento/trento-web</literal> and restart the
```
Comment (tomschr) [inline trento/xml/sso-integration.xml:69]: This should be a filename:

```suggestion
          <filename>/etc/trento/trento-web</filename> and restart the
```

File: trento/xml/sso-integration.xml [hunk 4]
```diff
@@ -0,0 +1,762 @@
...
+        </varlistentry>
+        <varlistentry>
+          <term>
+            OIDC_CALLBACK_URL
+          </term>
+          <listitem>
+            <para>
+              OIDC callback url where the IDP is redirecting once the
+              authentication is completed (default value:
+              <literal>https://#{TRENTO_WEB_ORIGIN}/auth/oidc_callback</literal>)
```
Comment (tomschr) [inline trento/xml/sso-integration.xml:153]: This should be an `<uri>`:

```suggestion
              <uri>https://#{TRENTO_WEB_ORIGIN}/auth/oidc_callback</uri>)
```

File: trento/xml/sso-integration.xml [hunk 5]
```diff
@@ -0,0 +1,762 @@
...
+      <para>
+        OAuth 2.0 authentication is <emphasis role="strong">disabled by
+        default</emphasis>.
+      </para>
+      <section xml:id="enabling-oauth-20-when-using-rpm-packages">
+        <title>Enabling OAuth 2.0 when using RPM packages</title>
+        <para>
+          Provide the following environment variables to trento-web
+          configuration, which is stored at
+          <literal>/etc/trento/trento-web</literal> and restart the
```
Comment (tomschr) [inline trento/xml/sso-integration.xml:177]: Should be a `<filename>`:

```suggestion
          <filename>/etc/trento/trento-web</filename> and restart the
```

File: trento/xml/sso-integration.xml [hunk 6]
```diff
@@ -0,0 +1,762 @@
...
+OAUTH2_CALLBACK_URL=&lt;&lt;OAUTH2_CALLBACK_URL&gt;&gt;
+</programlisting>
+        <blockquote>
+          <para>
+            <emphasis role="strong">Note:</emphasis> OAUTH2_SCOPES is an
+            optional variable with the default value
+            <literal>openid profile email</literal>. OAUTH2_SCOPES must
+            be adjusted depending on IDP provider requirements.
+          </para>
+        </blockquote>
```
Comment (tomschr) [inline trento/xml/sso-integration.xml:201]: This is wrong. If you want a note, use this:

```suggestion
        <note>
          <para>
            OAUTH2_SCOPES is an
            optional variable with the default value
            <literal>openid profile email</literal>. OAUTH2_SCOPES must
            be adjusted depending on IDP provider requirements.
          </para>
        </note>
```

File: trento/xml/sso-integration.xml [hunk 7]
```diff
@@ -0,0 +1,762 @@
...
+          (Trento in this case) must be provided to the IDP, the
+          Certificate file in this case.
+        </para>
+        <para>
+          For this reason, Trento already provides a certificates set
+          created during the installation. When Trento is installed the
+          first time (does not matter the installation mode) the
+          certificates are created, and the public certificate file
+          content is available in the
+          <literal>http://localhost:4000/api/public_keys</literal>
```
Comment (tomschr) [inline trento/xml/sso-integration.xml:384]: This should be an `<uri>`:

```suggestion
          <uri>http://localhost:4000/api/public_keys</uri>
```

File: trento/xml/sso-integration.xml [hunk 8]
```diff
@@ -0,0 +1,762 @@
...
+curl http://localhost:4000/api/public_keys
+</programlisting>
+        <para>
+          Copy the content of the certificate from there, and provide it
+          to the IDP. This way, the IDP will sign and verify the
+          messages sent by both ends.
+        </para>
+        <para>
+          When this certificate is used, and provided to the IDP, the
+          IDP recreates its own <literal>metadata.xml</literal> file.
```
Comment (tomschr) [inline trento/xml/sso-integration.xml:397]:
```suggestion
          IDP recreates its own <filename>metadata.xml</filename> file.
```

File: trento/xml/sso-integration.xml [hunk 9]
```diff
@@ -0,0 +1,762 @@
...
+          This file defines which certificate is used to sign the
+          messages by both sides. At this point, Trento Web must be
+          restarted to use the new <literal>metadata.xml</literal>
```
Comment (tomschr) [inline trento/xml/sso-integration.xml:400]:
```suggestion
          restarted to use the new <filename>metadata.xml</filename>
```

File: trento/xml/sso-integration.xml [hunk 10]
```diff
@@ -0,0 +1,762 @@
...
+          content.
+        </para>
+        <para>
+          If the <literal>SAML_METDATA_CONTENT</literal> option is being
```
Comment (tomschr) [inline trento/xml/sso-integration.xml:404]:
```suggestion
          If the <option>SAML_METDATA_CONTENT</option> option is being
```

File: trento/xml/sso-integration.xml [hunk 11]
```diff
@@ -0,0 +1,762 @@
...
+          used, the content of this variable must be updated with the
+          new metadata. In the other hand, if
+          <literal>SAML_METADATA_URL</literal> is used, the new metadata
```
Comment (tomschr) [inline trento/xml/sso-integration.xml:407]:
```suggestion
          <option>SAML_METADATA_URL</option> is used, the new metadata
```

File: trento/xml/sso-integration.xml [hunk 12]
```diff
@@ -0,0 +1,762 @@
...
+          completed, communication will fail because the message
+          signatures will not be recognized
+        </para>
+        <para>
+          <emphasis role="strong">This restart must be done manually, by
+          the admin. If the installation is done by a
+          <literal>RPM</literal>, restarting the
+          <literal>systemd</literal> daemon. If the installation is done
+          using <literal>Docker</literal>, the container must be
+          restarted.</emphasis>
```
Comment (tomschr) [inline trento/xml/sso-integration.xml:418]: I limit the emphasis to the core sentence, not the whole paragraph.

```suggestion
          <emphasis role="strong">This restart must be done manually, by
          the admin.</emphasis> If the installation is done by a
          <literal>RPM</literal>, restarting the
          <literal>systemd</literal> daemon. If the installation is done
          using <literal>Docker</literal>, the container must be
          restarted.</emphasis>
```

If you need a more visible appearance, use `<note>` or even `<important>`.

File: trento/xml/sso-integration.xml [hunk 13]
```diff
@@ -0,0 +1,762 @@
...
+        </varlistentry>
+        <varlistentry>
+          <term>
+            SAML_SP_DIR
+          </term>
+          <listitem>
+            <para>
+              SAML SP directory, where SP specific required files (such
+              as certificates and metadata file) are placed (default
+              value: <literal>/etc/trento/trento-web/saml</literal>)
```
Comment (tomschr) [inline trento/xml/sso-integration.xml:595]:
```suggestion
              value: <filename>/etc/trento/trento-web/saml</filename>)
```

File: trento/xml/sso-integration.xml [hunk 14]
```diff
@@ -0,0 +1,762 @@
...
+        </varlistentry>
+        <varlistentry>
+          <term>
+            SAML_IDP_NAMEID_FORMAT
+          </term>
+          <listitem>
+            <para>
+              SAML IDP name id format, used to interpret the attribute
+              name. Whole urn string must be used (default value:
+              <literal>SAML IDP name id format, used to interpret the attribute name. Whole urn string must be used</literal>)
```
Comment (tomschr) [inline trento/xml/sso-integration.xml:583]: This will become a render problem as `<literal>` does not add linebreaks. Use `<screen>` instead if you really want to have this whole long string.

**Review by tomschr — CHANGES_REQUESTED** [review]
@arbulu89 Thanks for the submission. :+1: 

I've reviewed it from a DocBook perspective. :wink: Here is a summary:

* Some tags should be renamed. As Markdown is a "stupid" format, it cannot distinguish between a filename and an option. I marked it in some lines.
* I haven't looked at the text itself yet. :smile: This is something for another round when all the structure is in place.
* Be careful with `<emphasise role="strong">`. We use it rarely in our text. This could be fixed in the original Markdown source.

File: trento/xml/sso-integration.xml (hunk 14 above)
Comment (arbulu89) [reply trento/xml/sso-integration.xml:583]: This is actually a copy/paste error. It will be changed by other value

**Comment by arbulu89** [comment]
Hey @tomschr 
I think I have applied all the corrections. As I still want to have the markdown as source of truth, I have done some "nasty" things there. It doesn't look that much as a valid markdown, but this is the price we need to pay to have a correctly generated xml file

**Comment by abravosuse** [comment]
@tomschr once both the structure and the content are review, could you please generate pdf file so that I can review the PR as well? Thanks!

File: trento/migration/sso-integration.md [hunk 15]
```diff
@@ -0,0 +1,390 @@
...
+The user profile must include attributes for: username, email, first name and last name.
+This attributes must be mapped for all users wanting to connect to Trento.
+
+By default, Trento expects the `username`, `email`, `firstName` and `lastName` attribute names. All these 4 attribute names are configurable using the next environment variables, following the same order: `SAML_USERNAME_ATTR_NAME`, `SAML_EMAIL_ATTR_NAME`, `SAML_FIRSTNAME_ATTR_NAME` and `SAML_LASTNAME_ATTR_NAME`.
+So for example, if the IDP user profile username is defined as `attr:username`, `SAML_USERNAME_ATTR_NAME=attr:username` should be used.
+
+#### Signing keys
+
+Commonly, SAML protocol messages are signed with SSL. This is optional using Trento, and the signing is not required (even though it is recommended).
+If the IDP signs the messages, and expect signed messages back, certificates used by the SP (Trento in this case) must be provided to the IDP, the Certificate file in this case.
```
Comment (tomschr) [inline trento/migration/sso-integration.md:202]: What's "SP" here? Service Provider? I've splitted the sentence a bit. But I'm not sure how to use the "the certificate file in this case". Could you clarify that?

```suggestion
If the IDP signs the messages and expect signed messages back. Certificates used by the Trento service provider must be provided to the IDP, the certificate file in this case.
```

File: trento/migration/sso-integration.md [hunk 16]
```diff
@@ -0,0 +1,390 @@
...
+
+For this reason, Trento already provides a certificates set created during the installation. When Trento is installed the first time (does not matter the installation mode) the certificates are created, and the public certificate file content is available in the <uri>http://localhost:4000/api/public_keys</uri> route. 
+
```
Comment (tomschr) [inline trento/migration/sso-integration.md:205]:
```suggestion
For this reason, Trento already provides a certificates set created during the installation. When Trento is installed the first time (does not matter the installation mode) the certificates are created, and the public certificate file content is available in the <uri>http://localhost:4000/api/public_keys</uri> route.

Use the following command to download the certificate:

```

**Review by tomschr — CHANGES_REQUESTED** [review]
Thanks @arbulu89 for all your efforts! Much appreciated! I had the time to review it now. :slightly_smiling_face: 

Before you start working on the comments itself (maybe they aren't that important), we should maybe discuss about the overall structure. Sometimes I was a bit confused.

The Trento documentation is structured in a way that it distinguishes between two types:

* **Tasks** that the user has to execute to achieve a specific goal.
* **Concepts** that explains background information.

It would be helpful if your text could be adapted to this "doc framework"/convention. At the moment, I see a mixture of both. Don't worry, I'm here to help you! :slightly_smiling_face: 

If you look through the Trento documentation, you can see lots of tasks ("_Installing_ Trento Server" instead of "Trento Server Installation"). 

When Alberto and I described a task, we use a _procedure_. A procedure is a step-by-step instruction what the user has to do.

So my more holistic suggestion would be:

1. What's a task and what is a concept in your text?
2. If it's a task, what steps do you need? Can we rewrite the text into some steps?

So your text is about single sign-on. Just from my gut feeling, we could discuss if this structure would be appropriate:

```
4.5 Integrate Single Sign-On
  4.5.1 Background information
        Tell here about available protocols, user roles, and
        authentication. Maybe also requirements?
        Don't add tasks.
  4.5.2 Using/Integrating OpenID Connect
  4.5.3 Using/Integrating OAuth 2
  4.5.4 Using/Integrating Security Assertion Markup Language
```

To make it a bit more visible, this is how it could look like if you write a procedure for section 4.5.2


```
# Using/Integrating OpenID Connect
By default, OIDC is disabled. To enable OIDC in Trento, proceed
as follows:

1. Open the file /etc/trento/trento-web
2. Provide the following environment variables
   # add here a short description
3. Restart the application with:
   # add here the command
```

Hope that makes it clearer what I intend. :slightly_smiling_face:

File: trento/migration/sso-integration.md [hunk 17]
```diff
@@ -0,0 +1,390 @@
...
+Trento integrates with an IDP that uses the OIDC protocol to authenticate users accessing the Trento web console.
+
+### Enabling OIDC
+
+OIDC authentication is **disabled by default**.
+
+#### Enabling OIDC when using RPM packages
+
+Provide the following environment variables to trento-web configuration, which is stored at <filename>/etc/trento/trento-web</filename> and restart the application to enable OIDC integration.
```
[the remark this answers is not in the document]
Comment (arbulu89) [reply trento/migration/sso-integration.md]: You already rearranged this, and it looks fine

File: trento/migration/sso-integration.md [hunk 18]
```diff
@@ -0,0 +1,390 @@
...
+OIDC_CLIENT_ID=<<OIDC_CLIENT_ID>>
+OIDC_CLIENT_SECRET=<<OIDC_CLIENT_SECRET>>
+OIDC_BASE_URL=<<OIDC_BASE_URL>>
+
+# Optional:
+OIDC_CALLBACK_URL=<<OIDC_CALLBACK_URL>>
+```
+#### Enabling OIDC when using Docker images
+
+Provide the following environment variables to the docker container and restart the application to enable OIDC integration.
```
[the remark this answers is not in the document]
Comment (arbulu89) [reply trento/migration/sso-integration.md]: Changed by you XD

File: trento/migration/sso-integration.md [hunk 19]
```diff
@@ -0,0 +1,390 @@
...
+
+Trento integrates with an IDP that uses the OAuth 2 protocol to authenticate users accessing the Trento web console.
+
+### Enabling OAuth 2.0
+
+OAuth 2.0 authentication is **disabled by default**.
+
+#### Enabling OAuth 2.0 when using RPM packages
+
+Provide the following environment variables to trento-web configuration, which is stored at <filename>/etc/trento/trento-web</filename> and restart the application to enable OAuth 2.0 integration.
```
[the remark this answers is not in the document]
Comment (arbulu89) [reply trento/migration/sso-integration.md]: Same, already changed by you

File: trento/migration/sso-integration.md (hunk 15 above)
Comment (arbulu89) [reply trento/migration/sso-integration.md:237]: Let me rephrase this and move it to the top chapter, as a background information

File: trento/migration/sso-integration.md [hunk 20]
```diff
@@ -0,0 +1,441 @@
...
+This attributes must be mapped for all users wanting to connect to Trento.
+
+By default, Trento expects the `username`, `email`, `firstName` and `lastName` attribute names. All these 4 attribute names are configurable using the next environment variables, following the same order: `SAML_USERNAME_ATTR_NAME`, `SAML_EMAIL_ATTR_NAME`, `SAML_FIRSTNAME_ATTR_NAME` and `SAML_LASTNAME_ATTR_NAME`.
+So for example, if the IDP user profile username is defined as `attr:username`, `SAML_USERNAME_ATTR_NAME=attr:username` should be used.
+
+#### SAML redirect URI
+
+Once the login is done succesfully, the IDP redirects the session back to Trento. This redirection is done to `https://trento.example.com/sso/sp/consume/saml`, so this URI must be set as valid in the IDP.
+
+### Get certificates from Trento
```
Comment (tomschr) [inline trento/migration/sso-integration.md:240]:
```suggestion
### Downloading certificates from Trento
```

File: trento/migration/sso-integration.md [hunk 21]
```diff
@@ -0,0 +1,441 @@
...
+
+Trento provides a certificates set created during the installation. When Trento is installed the first time (does not matter the installation mode) the certificates are created, and the public certificate file content is available in the <uri>http://localhost:4000/api/public_keys</uri> route.
```
Comment (tomschr) [inline trento/migration/sso-integration.md:242]: I would rephrase it a bit:

```suggestion
Trento provides a certificates set created during the installation. Regardless of the installation mode, when Trento is installed the first time the certificates are created and the public certificate file content is available in the <uri>http://localhost:4000/api/public_keys</uri> route. 
```

File: trento/migration/sso-integration.md [hunk 22]
```diff
@@ -0,0 +1,441 @@
...
+
+Trento provides a certificates set created during the installation. When Trento is installed the first time (does not matter the installation mode) the certificates are created, and the public certificate file content is available in the <uri>http://localhost:4000/api/public_keys</uri> route. 
+
+Use the following command to download the certificate:
+
+```bash
+curl http://localhost:4000/api/public_keys
+```
+    
+Copy the content of the certificate from there, and provide it to the IDP. This way, the IDP will sign and verify the messages sent by both ends.
```
Comment (tomschr) [inline trento/migration/sso-integration.md:250]: What do you mean by "provide it to the IDP"? And "both ends" means Trento and IDP, right?

```suggestion
Copy the content of the certificate from there and provide it to the IDP. This way, the IDP will sign and verify the messages sent by both ends.
```

File: trento/migration/sso-integration.md [hunk 23]
```diff
@@ -0,0 +1,441 @@
...
+SAML_SP_ORG_URL=<SAML_SP_ORG_URL>
+SAML_USERNAME_ATTR_NAME=<SAML_USERNAME_ATTR_NAME>
+SAML_EMAIL_ATTR_NAME=<SAML_EMAIL_ATTR_NAME>
+SAML_FIRSTNAME_ATTR_NAME=<SAML_FIRSTNAME_ATTR_NAME>
+SAML_LASTNAME_ATTR_NAME=<SAML_LASTNAME_ATTR_NAME>
+SAML_SIGN_REQUESTS=<SAML_SIGN_REQUESTS>
+SAML_SIGN_METADATA=<SAML_SIGN_METADATA>
+SAML_SIGNED_ASSERTION=<SAML_SIGNED_ASSERTION>
+SAML_SIGNED_ENVELOPES=<SAML_SIGNED_ENVELOPES>
+```
```
Comment (tomschr) [inline trento/migration/sso-integration.md:314]: This needs to be intended as it belongs under the item.

**Review by tomschr — CHANGES_REQUESTED** [review]
@arbulu89 Thanks for the efforts. Here is my suggestion. :wink:

File: trento/migration/sso-integration.md [hunk 24]
```diff
@@ -0,0 +1,441 @@
...
+To use an existing SAML IDP, follow the next instrunctions to met the specific requirements. You need:
+
+* Configure SAML IDP and user profiles
+* Start Trento to generate the certificates and download them
+* Provide the generated certificate to the IDP
+* Restart Trento
+
+### Configuring SAML IDP setup
+
+Configure the existing IDP with the next minimum options to be able to connect with Trento's Service Provider (SP).
```
[the remark this answers is not in the document]
Comment (arbulu89) [reply trento/migration/sso-integration.md:225]: Yes...
Thank you

File: trento/migration/sso-integration.md (hunk 20 above)
Comment (arbulu89) [reply trento/migration/sso-integration.md:240]: I'm going with `Getting` as you get the content as a string, not a file itself

File: trento/migration/sso-integration.md (hunk 22 above)
Comment (arbulu89) [reply trento/migration/sso-integration.md:250]: No, Trento already have it (it was created by Trento), so you need to copy the content and provide to the IDP, and we cannot say how you do that, because this depends on the IDP. But all IDPs will have the option to use it

Comment (arbulu89) [reply trento/migration/sso-integration.md:250]: I will rephrase to `Copy the content of the certificate from there and provide it to the IDP. This way, the IDP will sign its messages and verify the messages received from Trento.`

File: trento/migration/sso-integration.md [hunk 25]
```diff
@@ -0,0 +1,441 @@
...
+curl http://localhost:4000/api/public_keys
+```
+    
+Copy the content of the certificate from there, and provide it to the IDP. This way, the IDP will sign and verify the messages sent by both ends.
+
+### Restart Trento
+
+Once the certificate is provided to the IDP, the IDP recreates its own <filename>metadata.xml</filename> file. This file defines which certificate is used to sign the messages by both sides. At this point, Trento Web must be restarted to use the new <filename>metadata.xml</filename> content.
+
+If the <option>SAML_METDATA_CONTENT</option> option is being used, the content of this variable must be updated with the new metadata. In the other hand, if <option>SAML_METADATA_URL</option> is used, the new metadata is automatically fetched. If neither of these steps are completed, communication will fail because the message signatures will not be recognized
```
[the remark this answers is not in the document]
Comment (arbulu89) [reply trento/migration/sso-integration.md:256]: Yes, once you get the certificate from trento, you provide it to the IDP (each IDP will have its own way).
Now, the IDP will update its own IDP with the given certificate.
Some IDP has a endpoint that you can access using a url that returns you the content of the metadata. 
Others don't, and it just gives you some other way (plain text) for example. 

If you are in the 1st option (IDP has metadata endpoint), the easiest way is to use it (as if for some reason, you change something in the IDP, Trento will fetch the update automatically).

In the second case (IDP doesn't have an endpoint), you will need to copy the content of the metadata.xml (it is given as one lines string) in the `SAML_METADATA_CONTENT` variable.

File: trento/migration/sso-integration.md [hunk 26]
```diff
@@ -0,0 +1,441 @@
...
+Configure the existing IDP with the next minimum options to be able to connect with Trento's Service Provider (SP).
+
+#### SAML user profile
+
+Users provided by the SAML installation must have some few mandatory attributes to login in Trento. All of them are mandatory, even though their name is configurable.
+The user profile must include attributes for: username, email, first name and last name.
+This attributes must be mapped for all users wanting to connect to Trento.
+
+By default, Trento expects the `username`, `email`, `firstName` and `lastName` attribute names. All these 4 attribute names are configurable using the next environment variables, following the same order: `SAML_USERNAME_ATTR_NAME`, `SAML_EMAIL_ATTR_NAME`, `SAML_FIRSTNAME_ATTR_NAME` and `SAML_LASTNAME_ATTR_NAME`.
+So for example, if the IDP user profile username is defined as `attr:username`, `SAML_USERNAME_ATTR_NAME=attr:username` should be used.
```
[the remark this answers is not in the document]
Comment (arbulu89) [reply trento/migration/sso-integration.md:234]: Ok, this is how SAML works.
When you have a SAML IDP, you have a user profile (one) that is used to model all the users created in the IDP.
This user profile, has some fields, the typical ones are `email`, `name`, etc

Each IDP installation will have different naming for the fields though. Some might have `email`, others `userEmail`, others `attr:email`, etc

So what the users needs to do is:
1. Have the attributes in the user profile (which the name they prefer)
2. Configure trento with those names

Now, that I have listed you this, I will transform the content to something similar

**Comment by arbulu89** [comment]
Hey @tomschr ,
I have some more improvements in the SAML chapter.
So far, I see 2 main points we might want to discuss.

The `Configuring SAML user profile` and `Restarting Trento` chapters. I have improved both. Let's see if you like them more

File: trento/migration/sso-integration.md [hunk 27]
```diff
@@ -0,0 +1,451 @@
...
+
+Once the login is done succesfully, the IDP redirects the session back to Trento. This redirection is done to `https://trento.example.com/sso/sp/consume/saml`, so this URI must be set as valid in the IDP.
+
+### Restarting Trento
+
+Once the certificate is provided to the IDP, the IDP recreates its own <filename>metadata.xml</filename> file. This file defines which certificate is used to sign the messages by both sides. At this point, Trento Web must be restarted to use the new <filename>metadata.xml</filename> content.
+
+If the <option>SAML_METDATA_CONTENT</option> option is being used, the content of this variable must be updated with the new metadata, as single line string. In the other hand, if <option>SAML_METADATA_URL</option> is used, the new metadata is automatically fetched. If neither of these steps are completed, communication will fail because the message signatures will not be recognized.
+
+If the used IDP has the endpoint to provide the <filename>metadata.xml</filename> file content, the usage of <option>SAML_METADATA_URL</option> is preferred, as further changes in the IDP will be fetched automatically by Trento when it is restarted.
```
Comment (tomschr) [inline trento/migration/sso-integration.md:266]:
```suggestion
If the used IDP has the endpoint to provide the <filename>metadata.xml</filename> file content, prefer the variable <option>SAML_METADATA_URL</option>. Trento will automatically fetch metadata when restarted.
```

File: trento/migration/sso-integration.md [hunk 28]
```diff
@@ -0,0 +1,451 @@
...
+
+```{=docbook}
+<note>
+  <para>This restart must be done manually, by the admin.</para>
+</note>
+```
+
+Follow the next instrucionts to restart with the configured options:
```
Comment (tomschr) [inline trento/migration/sso-integration.md:274]: Typo:

```suggestion
Follow the next instructions to restart with the configured options:
```

**Review by tomschr — CHANGES_REQUESTED** [review]
Thanks @arbulu89. Some smaller things.

File: trento/migration/sso-integration.md [hunk 29]
```diff
@@ -0,0 +1,447 @@
...
+
+So both the IDP and Trento must know how these 4 fields are mapped. For that, follow the next instruction:
+
+1. Add the attributes if they don't exist in the IDP user profile. If they already exist, they name doesn't need to be changed, so the current values can continue being used.
+
+1. Configure Trento to use the IDP attribute field names. For that, set the `SAML_USERNAME_ATTR_NAME`, `SAML_EMAIL_ATTR_NAME`, `SAML_FIRSTNAME_ATTR_NAME` and `SAML_LASTNAME_ATTR_NAME` environment values with the values configured in the IDP. For example, if the IDP user profile username is defined as `attr:username`, `SAML_USERNAME_ATTR_NAME=attr:username` should be used.
+
+#### Checking SAML redirect URI
+
+Once the login is done succesfully, the IDP redirects the session back to Trento. This redirection is done to `https://trento.example.com/sso/sp/consume/saml`, so this URI must be set as valid in the IDP.
```
[the remark this answers is not in the document]
Comment (arbulu89) [reply trento/migration/sso-integration.md:258]: It looks really appropriate!

**Comment by tomschr** [comment]
> A pity that I cannot commit here, as I need to create the final xml

Argh, that's really pity, but maybe I can do that or Eugen (if he still has the permissions). :slightly_smiling_face: 

Thank you!

**Comment by EMaksy** [comment]
Hey @abravosuse  i uploaded the pdf, have a look for [review purposes](https://github.com/SUSE/doc-unversioned/blob/4bd52c92cf943d2adb21d9821ee9eb65048ea164/trento/pdf/SLES-SAP-trento_en.pdf)!

**Comment by abravosuse** [comment]
I plan to use the document to test the integration of my rolling environment with our internal IDP provider. But my rolling environment runs on K3s and the documentation seems to be specific to rpm and container deployment. Is the SSO integration not possible when Trento runs on a Kubernetes cluster?

**Comment by abravosuse** [comment]
> > I plan to use the document to test the integration of my rolling environment with our internal IDP provider. But my rolling environment runs on K3s and the documentation seems to be specific to rpm and container deployment. Is the SSO integration not possible when Trento runs on a Kubernetes cluster?
> 
> Yes, it is usable. The thing is, we don't have any real documentation about almost any variable used in the k3s deployment before hand, so I guess we didn't add them when Eugen started first to document the SSO integrations.
> 
> Do you want me to add them? At the end, it is just a bunch more of variables to be used with `--set`. Anyway, we don't have them in our internal documentation neither.

Yes, please. It'd be helpful to have at least the list of helm variables to be used to enable SSO integration with the different protocols.

**Comment by arbulu89** [comment]
@abravosuse Kubernetes deployment steps added in last commit.
@EMaksy Could you create the pdf again so it has the latest changes?

**Comment by EMaksy** [comment]
Here the latest changes: [SLES-SAP-trento_en.pdf](https://github.com/user-attachments/files/17360627/SLES-SAP-trento_en.pdf)

Check it out @abravosuse

File: trento/xml/sso-integration.xml [hunk 30]
```diff
@@ -0,0 +1,1012 @@
...
+</programlisting>
+          <para>
+            Additionally, the following optional values are available:
+          </para>
+          <programlisting>
+HELM_EXPERIMENTAL_OCI=1 helm ... \
+   --set trento-web.saml.enabled=true \
+   --set trento-web.saml.idpId=&lt;SAML_IDP_ID&gt; \
+   --set trento-web.saml.spId=&lt;SAML_SP_ID&gt; \
+   --set trento-web.saml.metadataUrl=&lt;SAML_METADATA_URL&gt; \
```
Comment (abravosuse) [inline trento/xml/sso-integration.xml:645]:
```suggestion
```
Remove these parameters as they are mandatory, not optional.

File: trento/xml/sso-integration.xml [hunk 31]
```diff
@@ -0,0 +1,1012 @@
...
+          </para>
+          <programlisting>
+HELM_EXPERIMENTAL_OCI=1 helm ... \
+   --set trento-web.saml.enabled=true \
+   --set trento-web.saml.idpId=&lt;SAML_IDP_ID&gt; \
+   --set trento-web.saml.spId=&lt;SAML_SP_ID&gt; \
+   --set trento-web.saml.metadataUrl=&lt;SAML_METADATA_URL&gt;
+</programlisting>
+          <para>
+            To use the <option>SAML_METDATA_CONTENT</option> option
```
Comment (abravosuse) [inline trento/xml/sso-integration.xml:627]:
```suggestion
            To use the <option>SAML_METADATA_CONTENT</option> option
```

File: trento/xml/sso-integration.xml [hunk 32]
```diff
@@ -0,0 +1,1012 @@
...
+          <programlisting>
+HELM_EXPERIMENTAL_OCI=1 helm ... \
+   --set trento-web.oauth2.enabled=true \
+   --set trento-web.oauth2.clientId=&lt;OAUTH2_CLIENT_ID&gt; \
+   --set trento-web.oauth2.clientSecret=&lt;OAUTH2_CLIENT_SECRET&gt; \
+   --set trento-web.oauth2.baseUrl=&lt;OAUTH2_BASE_URL&gt; \
+   --set trento-web.oauth2.authorizeUrl=&lt;OAUTH2_AUTHORIZE_URL&gt; \
+   --set trento-web.oauth2.tokenUrl=&lt;OAUTH2_TOKEN_URL&gt; \
+   --set trento-web.oauth2.userUrl=&lt;OAUTH2_USER_URL&gt; \
+   --set trento-web.oauth2.scopes=&lt;OAUTH2_SCOPES&gt;
```
Comment (abravosuse) [inline trento/xml/sso-integration.xml:253]: The MD file describes the scopes parameter as optional:
"trento-web.oauth2.scopes</option> variable is optional with `profile email` as default value"
Maybe we should do it here as well?

File: trento/migration/sso-integration.md [hunk 33]
```diff
@@ -0,0 +1,539 @@
...
+   ```
+   HELM_EXPERIMENTAL_OCI=1 helm ... \
+      --set trento-web.oauth2.enabled=true \
+      --set trento-web.oauth2.clientId=<OAUTH2_CLIENT_ID> \
+      --set trento-web.oauth2.clientSecret=<OAUTH2_CLIENT_SECRET> \
+      --set trento-web.oauth2.baseUrl=<OAUTH2_BASE_URL> \
+      --set trento-web.oauth2.authorizeUrl=<OAUTH2_AUTHORIZE_URL> \
+      --set trento-web.oauth2.tokenUrl=<OAUTH2_TOKEN_URL> \
+      --set trento-web.oauth2.userUrl=<OAUTH2_USER_URL> \
+      --set trento-web.oauth2.scopes=<OAUTH2_SCOPES>
```
Comment (abravosuse) [inline trento/migration/sso-integration.md:149]:
```suggestion
```
If the scopes parameter is optional, maybe we should remove it from here?

File: trento/migration/sso-integration.md [hunk 34]
```diff
@@ -0,0 +1,539 @@
...
+   ```
+
+   Additionally, the following optional values are available:
+
+   ```
+   HELM_EXPERIMENTAL_OCI=1 helm ... \
+      --set trento-web.saml.enabled=true \
+      --set trento-web.saml.idpId=<SAML_IDP_ID> \
+      --set trento-web.saml.spId=<SAML_SP_ID> \
+      --set trento-web.saml.metadataUrl=<SAML_METADATA_URL> \
```
Comment (abravosuse) [inline trento/migration/sso-integration.md:351]:
```suggestion
```
These are mandatory parameters and therefore should not be mentioned here.

**Review by abravosuse — CHANGES_REQUESTED** [review]
Just a few minor change requests.

**Comment by abravosuse** [comment]
@arbulu89 I just left a couple of comments for your review.

File: trento/xml/sso-integration.xml (hunk 32 above)
Comment (arbulu89) [reply trento/xml/sso-integration.xml:253]: The `.md` and `.xml` files should have the exact same content. You don't need to review both.
Just have a look in the final `.xml`. If there is something wrong there, I will make changes in the `.md` file which are (should be at least, if I don't forget) translated to the final result

**Comment by arbulu89** [comment]
@abravosuse Thank you for this final review. All suggested changes done!
Just as a side note, you don't need to review both `.md` and `.xml` files. Just reviewing the last is enough for me, as anything wrong there, must be first changed in t he `md` file which will forward all the changes to the final result

**Review by abravosuse — APPROVED** [review]
Thank you @arbulu89 !

**Review by tomschr — APPROVED** [review]
